### PR TITLE
Adds documentation for WordPress.WP.EnqueueResourceParameters.

### PR DESCRIPTION
**Pull request by NielsdeBlaauw**
Related to #1722

### CODE REVIEW

File: WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml
Comment (jrfnl) [inline WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml]:
```suggestion
    The resource version must be set, to prevent the browser from retrieving an outdated, cached version after the resource has been updated.
```

File: WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml
Comment (jrfnl) [inline WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml]: I'm wondering if the example would be clearer if the last two (optional) parameters in this _invalid_ example were not passed at all ?

I realize that it would make it awkward, or rather impossible for the `<em>` tags to be used in this example, but so be it.

File: WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml
Comment (jrfnl) [inline WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml]:
```suggestion
    You must explicitly set a javascript resource to load in the header or footer of your page.
    It is recommended to load these resources in the footer. Loading scripts in the header blocks parsing of the page and has a negative impact on load times. However, loading in the footer may break compatibility when other scripts rely on the resource to be available at any time. In that case, you can pass `false` to make it explicit that the script should be loaded in the header of the page.
```

(and yes, you can use line breaks in the `standard` description)

**Comment by jrfnl** [comment]
He @NielsdeBlaauw Just checking in to see what the status is of this PR. Will you have a chance to finish this in the near future ?

**Comment by NielsdeBlaauw** [comment]
Hey @jrfnl 

At the moment I don't have the spare time to work on this. I'm fine with someone else picking this up, or will be working on it again in january.

File: WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml
Comment (NielsdeBlaauw) [reply WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml]: I've removed the null, indeed seems clearer this way, and the `<em>` is already in the valid example. Should be clear enough.

File: WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml
Comment (jrfnl) [inline WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml:4]:
```suggestion
    The resource version must be set, to prevent the browser from using an outdated, cached version after the resource has been updated.
```

The whole point is that the browser _won't_ be retrieving the new version, but will use the locally cached version.

**Review by jrfnl — DISMISSED** [review]
Hi @NielsdeBlaauw Thanks for the update and sorry it took so long for me to get to this.

I've had another look and just have some minor remarks:

* As the sniff looks at both the `register` and the `enqueue` variants of the functions, I feel it would be useful to show an example with those variants too.
* The description for the third part is very long, breaking it up into two paragraphs would improve the readability.
* Most example lines have trailing spaces.

As discussed via Slack DM and with your permission, I will update the PR to fix the above.

@GaryJones @dingo-d For whoever reviews next and merges this: please squash the commits on merge.

**Review by jrfnl — APPROVED** [review]

File: WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml
Comment (GaryJones) [inline WordPress/Docs/WP/EnqueuedResourceParametersStandard.xml:59]:
```suggestion
    You must explicitly set a JavaScript resource to load in either the header or the footer of your page. It is recommended to load these resources in the footer by setting the `$in_footer` parameter to `true`.
```

**Review by GaryJones — CHANGES_REQUESTED** [review]
Small typo fix needed.

**Review by dingo-d — APPROVED** [review]
I'll approve it so that, once the typo is fixed, Gary can merge it 🙂

**Review by GaryJones — APPROVED** [review]

**Pull request merged**
